### PR TITLE
Plugin to load hyper_v hostname from guest

### PR DESCRIPTION
**Pull request by safematix**
### Description

Plugin to load hyper_v hostname from guest, to see on which node the guest is running.

### Issues Resolved

- Issue #1302

### Check List

- [x] New functionality includes tests
- [x] All tests pass
- [ ] RELEASE\_NOTES.md, has been updated if required (not required for bugfixes, required for API changes)
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

**Comment by tas50** [comment]
There's a small chefstyle warning that you can resolve by running `chefstyle -a .` on the ohai dir

File: lib/ohai/plugins/linux/virtualization.rb [hunk 1]
```diff
@@ -150,6 +150,17 @@ def docker_exists?
       end
     end
 
+    # Detect Hyper-V gest and host hostname
```
Comment (tas50) [inline lib/ohai/plugins/linux/virtualization.rb]:
```suggestion
    # Detect Hyper-V guest and host hostname
```

File: lib/ohai/plugins/linux/virtualization.rb [hunk 2]
```diff
@@ -150,6 +150,17 @@ def docker_exists?
...
 
+    # Detect Hyper-V guest and the hostname of the host
+    if File.exist?("/var/lib/hyperv/.kvp_pool_3")
+      logger.trace("Plugin Virtualization: /var/lib/hyperv/.kvp_pool_3 contains string indicating Hyper-V guest")
+      data = File.read("/var/lib/hyperv/.kvp_pool_3")
+      hyperv_host = data[/\HostName(.*?)HostingSystemEditionId/, 1].scan(/[[:print:]]/).join.downcase
+      virtualization[:system] = "hyperv"
+      virtualization[:role] = "guest"
+      virtualization[:systems][:hyperv] = "guest"
+      virtualization[:host] = hyperv_host
```
Comment (tas50) [inline lib/ohai/plugins/linux/virtualization.rb]: I'm not sure how I feel about this attribute name. It seems a bit ambiguous. How would you feel about hypervisor_host instead? I'm open to other ideas as well.

Comment (safematix) [reply lib/ohai/plugins/linux/virtualization.rb]: That's a good idea, I think "hypervisor_host" is better than "host". That's more understandable and less confusing. :blush:

**Comment by tas50** [comment]
@safematix thanks for putting this together. This will be quite handy for a lot of users

**Pull request merged**